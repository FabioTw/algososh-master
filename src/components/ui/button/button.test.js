import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';

import { Button } from './button';

it('Кнопка рендерися без ошибок', () => {
  const buttonWithText = renderer.create(<Button text='test'/>).toJSON();
  const buttonWithOutText = renderer.create(<Button text=''/>).toJSON();
  const lockedButton = renderer.create(<Button text='test' disabled/>).toJSON();
  const buttonIsLoading = renderer.create(<Button isLoader={true}/>).toJSON();

  expect(buttonWithText).toMatchSnapshot();
  expect(buttonWithOutText).toMatchSnapshot();
  expect(lockedButton).toMatchSnapshot();
  expect(buttonIsLoading).toMatchSnapshot();
})

it('Нажатие на кнопку вызывает корректный alert', () => {
  window.alert = jest.fn();

  render(<Button text='test' type='button' isLoader={false} onClick={()=> alert('test was complited')}/>)

  const button = screen.getByText("test");
  fireEvent.click(button);

  expect(window.alert).toHaveBeenCalledWith('test was complited');
}); 