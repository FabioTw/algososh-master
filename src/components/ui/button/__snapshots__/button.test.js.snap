// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Кнопка рендерися без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;

exports[`Кнопка рендерися без ошибок 2`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    
  </p>
</button>
`;

exports[`Кнопка рендерися без ошибок 3`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  >
    test
  </p>
</button>
`;

exports[`Кнопка рендерися без ошибок 4`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
